instrumentation/vertx-web-3.0/testing/src/main/java/server/AbstractVertxWebServer.java

```diff
@@ -39,60 +39,46 @@ public Router buildRouter() {
             new Handler<RoutingContext>() {
               @Override
               public void handle(RoutingContext ctx) {
-                HttpServerTest.controller(
-                    SUCCESS,
-                    () -> {
-                      end(ctx.response().setStatusCode(SUCCESS.getStatus()), SUCCESS.getBody());
-                      return null;
-                    });
+                HttpServerTest.controller(SUCCESS, () -> null);
+                end(ctx.response().setStatusCode(SUCCESS.getStatus()), SUCCESS.getBody());
               }
             });
     router
         .route(INDEXED_CHILD.getPath())
         .handler(
-            ctx ->
-                HttpServerTest.controller(
-                    INDEXED_CHILD,
-                    () -> {
-                      INDEXED_CHILD.collectSpanAttributes(it -> ctx.request().getParam(it));
-                      end(ctx.response().setStatusCode(INDEXED_CHILD.getStatus()));
-                      return null;
-                    }));
+            ctx -> {
+              HttpServerTest.controller(
+                  INDEXED_CHILD,
+                  () -> {
+                    INDEXED_CHILD.collectSpanAttributes(it -> ctx.request().getParam(it));
+                    return null;
+                  });
+              end(ctx.response().setStatusCode(INDEXED_CHILD.getStatus()));
+            });
     router
         .route(QUERY_PARAM.getPath())
         .handler(
-            ctx ->
-                HttpServerTest.controller(
-                    QUERY_PARAM,
-                    () -> {
-                      end(
-                          ctx.response().setStatusCode(QUERY_PARAM.getStatus()),
-                          ctx.request().query());
-                      return null;
-                    }));
+            ctx -> {
+              HttpServerTest.controller(QUERY_PARAM, () -> null);
```

Comment:
WDYT about removing the second argument of `controller`? The passed lambda seems to be unneeded now